runtime/doc/lsp.txt

```diff
@@ -708,7 +708,7 @@ buf_request_sync({bufnr}, {method}, {params}, {timeout_ms})
 
     Return (multiple): ~
         (table) result Map of client_id:request_result.
-        (string|nil) err On timeout, cancel, or error, `err` is a string
+        string?|nil err On timeout, cancel, or error, `err` is a string
```

Comment:
looks like there are some problems with docgen now?

Reply:
Yes. I plan on looking into this tomorrow evening. I don't have time today due to work.

I don't know anything about docgen, so hopefully there is some flag to disable the manually appending `|nil` behavior.

Reply:
The docgen is an ad-hoc script that is developed just enough for it to work. You'll need to dig into the doxgen filter so it can handle `?`.

Reply:
I've made some changes in `lua2dox.lua` that fix these incorrect types.